### PR TITLE
[Contract] Implement Community-Based Review Moderation [Soroban]

### LINKED ISSUE #463: [Contract] Implement Community-Based Review Moderation [Soroban]
### 🎯 Objective
Develop a Soroban smart contract to enable community-based moderation of educational content reviews, ensuring high-quality feedback on the Stellar network.

### 🏗 Contract Structure
```rust
review-system/src/
  lib.rs                // Contract configuration and exports
  moderation.rs         // Community moderation logic
  utils.rs              // Shared utilities for voting and flagging
```

### 🗂 Requirements
1. **Core Functionality**
   - Allow community members to flag inappropriate reviews for moderation.
   - Implement a voting system for approving or rejecting flagged reviews.
2. **Additional Features**
   - Restrict moderation votes to verified Stellar account holders.
   - Emit events for flagging and moderation outcomes.
   - Integrate with user-reputation-contract to weight moderation votes by reputation.

### 📦 Key Data Structures
```rust
struct ModerationFlag {
    review_id: u64,        // Associated review ID
    flagger: Address,      // Stellar address of the flagger
    reason: String,        // Reason for flagging
    votes_approve: u32,    // Votes to approve removal
    votes_reject: u32,     // Votes to reject removal
}
```

### 🔑 Key Functions
- `flag_review(review_id: u64, reason: String)` – Flags a review for moderation.
- `vote_moderation(review_id: u64, approve: bool)` – Casts a vote on a flagged review.

### 🔗 References
- [Stellar Soroban Documentation](https://soroban.stellar.org/docs)
- [Rust Book](https://doc.rust-lang.org/book/)

### ⏳ ETA
- Estimated completion: 2 days.

### 📌 Additional Notes
- Ensure moderation votes are secure and resistant to manipulation.
- Test edge cases like multiple flags on the same review or vote spamming.
- Use standardized dispute resolution interfaces to avoid redundancy.

### PR DESCRIPTION
**Pull request by zachyo**
feat(soroban): Implement Community Review Moderation Contract

  Summary:

  This pull request delivers the complete implementation of the Soroban smart contract for community-based moderation of educational content reviews. The 
  work fulfills all core objectives and requirements outlined in the insstructions.md planning document.

  The contract establishes a decentralized system for flagging and voting on reviews, ensuring that feedback on the platform remains high-quality and free 
  from abuse.
  
  Closes #463 
  
## Completed Tasks:

  Based on the requirements document, the following features have been successfully implemented:

   - Core Functionality:
       - ✅ Implemented the flag_review function, allowing community members to flag inappropriate reviews for moderation.
       - ✅ Developed a complete voting system (vote_moderation) for the community to approve or reject flagged reviews.

   - Additional Features:
       - ✅ Integrated with an external user-reputation-contract to apply vote weighting, giving more influence to users with higher reputation scores.
       - ✅ Implemented on-chain events for all key actions (review_flagged, moderation_voted, moderation_resolved) to ensure full transparency.
       - ✅ Secured all state-changing functions with require_auth() to restrict actions to authenticated Stellar account holders.

   - Structure and Security:
       - ✅ The contract follows the proposed file structure (lib.rs, moderation.rs, utils.rs).
       - ✅ Implemented security measures to prevent manipulation, such as checking for and rejecting duplicate votes.
       - ✅ Added comprehensive unit tests to cover edge cases, including attempts to flag a review twice and vote spamming.

   - Documentation:
       - ✅ Created a detailed DOCUMENTATION.md file to ensure the contract is easy to unders

### CODE REVIEW

**Review by kimcascante — CHANGES_REQUESTED** [review]
@zachyo Please review CI errors

**Comment by zachyo** [comment]
@kimcascante . It has all been fixed

**Review by kimcascante — APPROVED** [review]
LGTM

File: .marscode/deviceInfo.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "deviceId": "7f91adcb12118e2c9981ba3eb11750e017f0b03a54e70479c08f7b5a3e2b3d76"
+}
```
Comment (salazarsebas) [inline .marscode/deviceInfo.json]: Hey @zachyo , please delete this file

**Review by salazarsebas — CHANGES_REQUESTED** [review]

**Comment by zachyo** [comment]
@salazarsebas Deleted.

**Review by salazarsebas — APPROVED** [review]
LGTM

**Review by xJeffx23 — APPROVED** [review]
